lib/puppet/file_serving/metadata.rb

```diff
@@ -118,6 +118,10 @@ def collect(source_permissions = nil)
     when "link"
       @destination = Puppet::FileSystem.readlink(real_path)
       @checksum = ("{#{@checksum_type}}") + send("#{@checksum_type}_file", real_path).to_s rescue nil
+    when "fifo"
+      @checksum = ("{#{@checksum_type}}") + send("#{@checksum_type}_file", real_path).to_s
```

Comment:
I don't think we can checksum/read the fifo, otherwise puppet will hang. For example, the `sha256_file` method calls: https://github.com/puppetlabs/puppet/blob/ef8d5bf2d6d72fff79e11e7322bcf3ce4bf1694f/lib/puppet/util/checksums.rb#L361-L362 Maybe best to handle this the `directory` case, where we "checksum" based on `ctime`? Like the same for `socket`?

Reply:
After some manual testing, it seems that, while recursing, `checksum_type` is by default set to `none` (and thus calling the `none_file` method which returns an empty string as checksum). For good measure, I've set it explicitly to `none` (even though the same seems to be happening in `file` cases) and everything seems to be working as expected.